src/image_ops.rs

```diff
@@ -1210,6 +1212,95 @@ cropping if it does not fit.",
         Example::new(1, 1, "DUP > const 184 268 (255, 255, 0) > overlay 10 50")
 );
 
+//-----------------------------------------------------------------------------
+// Pad
+//-----------------------------------------------------------------------------
+
+#[derive(Debug, Clone, PartialEq)]
+struct Pad {
+    top: u32,
+    right: u32,
+    bottom: u32,
+    left: u32,
+    color: Color,
+}
+
+impl ImageOp for Pad {
+    fn apply(&self, stack: &mut ImageStack) {
+        let over = stack.pop();
+
+        let width = self.left + self.right + over.width();
+        let height = self.top + self.bottom + over.height();
+
+        let under = match self.color {
+            Color::Luma(l) => ImageLuma8(ImageBuffer::from_pixel(width, height, l)),
+            Color::LumaA(l) => ImageLumaA8(ImageBuffer::from_pixel(width, height, l)),
+            Color::Rgb(l) => ImageRgb8(ImageBuffer::from_pixel(width, height, l)),
+            Color::Rgba(l) => ImageRgba8(ImageBuffer::from_pixel(width, height, l)),
+        };
+
+        use DynamicImage::*;
+        let result = match (under, over) {
+            (ImageLuma8(mut under), ImageLuma8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageLuma8(under)
+            }
+            (ImageLumaA8(mut under), ImageLumaA8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageLumaA8(under)
+            }
+            (ImageRgb8(mut under), ImageRgb8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageRgb8(under)
+            }
+            (ImageRgba8(mut under), ImageRgba8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageRgba8(under)
+            }
+            (ImageBgr8(mut under), ImageBgr8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageBgr8(under)
+            }
+            (ImageBgra8(mut under), ImageBgra8(over)) => {
+                image::imageops::overlay(&mut under, &over, self.left, self.top);
+                ImageBgra8(under)
+            }
+            // Due to the call to convert_to_color_space above
```

Comment:
This looks to be missing a call to `convert_to_color_space`. So if e.g. you attempt to pad an RGB image with a grayscale border you'll hit this branch.

The overlay operation converts the `over` image to the same format as the `under` image. In this case I think that converting the image if required makes the most sense (although the other way round isn't obviously worse). You can do this by adding `let over = convert_to_color(over, color_space(under))` immediately before this match. Please also document this behaviour in the description within `impl_parse` below.